test/fhirtypes/ElementDefinition.applyFlags.test.ts

```diff
@@ -19,7 +19,7 @@ describe('ElementDefinition', () => {
     fisher = new TestFisher().withFHIR(defs);
   });
   beforeEach(() => {
-    observation = fisher.fishForStructureDefinition('Observation');
+    observation = fisher.fishForStructureDefinition('us-core-observation-lab');
```

Comment:
'Observation' is a resource and 'us-core-observation-lab' is a profile... The resources causes the following failure so switched to a profile:.
`The MustSupport flag is not permitted on element Observation.note of Observation (allowed only in profiles and extensions)`

Reply:
@cmoesel @ngfreiter I actually wrote myself a note about this (but it got buried). Many of the existing tests are using resources rather than profiles for test StructureDefinitions. In many cases, the scope of the test means it shouldn't matter, but in the above case, it did. It might make some sense later on to maybe make the tests a little more "rigerous" by using the appropriate type of StructureDefinitions when testing things like rules. Now that custom resources and logical models are in the mix, it might be beneficial to ensure we are testing with objects that the rules are intended to operate on/with to prevent something from slipping the the cracks down the road.

Reply:
Ah.  Interesting.  I guess I had never noticed that we have tests that apply constraints to resources (rather than profiles).  I agree -- this is something we should try to avoid.  Where possible, our tests should represent at least _good_ practice, if not _best_ practice!

I logged this as #815.